lib/ansible/plugins/connection/ssh.py

```diff
@@ -850,10 +850,34 @@ def exec_command(self, cmd, in_data=None, sudoable=True):
             args = (ssh_executable, self.host, cmd)
 
         cmd = self._build_command(*args)
+
+        if display.verbosity >= 4:
+            # for 'ssh -G', the 'cmd' is not used, so this just removes it from the
+            # ssh command we use for a cleaner msg.
+            debug_args = args[:-1]
+            self._display_ssh_config(debug_args, in_data, sudoable)
+
         (returncode, stdout, stderr) = self._run(cmd, in_data, sudoable=sudoable)
 
         return (returncode, stdout, stderr)
 
+    def _display_ssh_config(self, args, in_data=None, sudoable=True):
+        # only 'ssh' (not sftp or scp) support -G. And only ~recentish versions.
+
+        debug_args = tuple(args[:] + ('-G',))
+        cmd = self._build_command(*debug_args)
+
+        try:
+            (returncode, stdout, stderr) = self._run(cmd, in_data, sudoable, checkrc=False)
+        except AnsibleConnectionFailure:
+            # -G failed, could be too old, not openssh, etc.
```

Comment:
That may depend on the version, but OpenSSH_7.4p1 shows the applied config, including command line options and -o options.

.ssh.config snippet
```
Host local_ssh_ssh_config_wrong_user
    HostName 127.0.0.1
    User wronguser
```


```
[fedora-25:ansible (facts_notes %)]$ ssh -G local_ssh_ssh_config_wrong_user | grep compress
compression no
compressionlevel 6
[fedora-25:ansible (facts_notes %)]$ ssh -G -C adrian@local_ssh_ssh_config_wrong_user | grep compress
compression yes
compressionlevel 6
[fedora-25:ansible (facts_notes %)]$ ssh -G local_ssh_ssh_config_wrong_user | grep user
user wronguser
userknownhostsfile ~/.ssh/known_hosts ~/.ssh/known_hosts2
[fedora-25:ansible (facts_notes %)]$ ssh -G adrian@local_ssh_ssh_config_wrong_user | grep user
user adrian
userknownhostsfile ~/.ssh/known_hosts ~/.ssh/known_hosts2
[fedora-25:ansible (facts_notes %)]$ ssh -G -o User=adrian local_ssh_ssh_config_wrong_user | grep user
user adrian
userknownhostsfile ~/.ssh/known_hosts ~/.ssh/known_hosts2
[fedora-25:ansible (facts_notes %)]$ man ssh
[fedora-25:ansible (facts_notes %)]$ ssh -G -l adrian local_ssh_ssh_config_wrong_user | grep user
user adrian
userknownhostsfile ~/.ssh/known_hosts ~/.ssh/known_hosts2
[fedora-25:ansible (facts_notes %)]$ ssh -G -l adrian local_ssh_ssh_config_wrong_user | grep agent
forwardagent no
[fedora-25:ansible (facts_notes %)]$ ssh -G -l adrian local_ssh_ssh_config_wrong_user -A | grep agent
forwardagent yes
```